feature/#126: implementing change of snake speed.

gabriel-vr opened the pull request:
The game delay can increased and decreased by the keys '+' and '-' while the game is being played or it is paused.
Fixes #126 .

If you find something wrong with this PR, leave a comment please. This is my first PR for the project.

gabriel-vr committed:
feature: implementing change of snake speed. The speed can be modified while the game is being played or paused.

lucasvianav commented on src/ttsnake.c:
```diff
@@ -580,6 +587,20 @@ void * userinput(){
 					game_end = 0;
 				}
 				break;
+      case '+':
+        if(!game_end && !restarted){
+          if(game_delay > MIN_GAME_DELAY){
+            game_delay-=DELTA_DELAY;
+          }
+        }
+        break;
+      case '-':
+        if(!game_end && !restarted){
+          if(game_delay < MAX_GAME_DELAY){
+            game_delay+=DELTA_DELAY;
+          }
+        }
+        break;
```
I'm not sure if it could be a problem for the current values of `MIN_GAME_DELAY`, `MAX_GAME_DELAY`, `DEFAULT_GAME_DELAY` and `DELTA_DELAY`, but I think it's safer to check if the "would-be value" of `game_delay` is higher/lower than the max/min, instead of the current value. 

What do you think of:

```suggestion
      case '+':
        if (!game_end && !restarted) {
          if(game_delay - DELTA_DELAY >= MIN_GAME_DELAY){
            game_delay -= DELTA_DELAY;
          } else {
            game_delay = MIN_GAME_DELAY;
          }
        }
        break;
      case '-':
        if(!game_end && !restarted){
          if (game_delay + DELTA_DELAY <= MAX_GAME_DELAY) {
            game_delay += DELTA_DELAY;
          } else {
            game_delay = MAX_GAME_DELAY;
          }
        }
        break;
```

gabriel-vr committed:
Update src/ttsnake.c

Checking if the possible new value of delay is not in the set of acceptable values, instead of checking if the actual value is not in this set.

Co-authored-by: Lucas Viana <oviana@outlook.com>

gabriel-vr commented:
Should i just merge it now or i must wait to someone do it?

lucasvianav commented:
> Should i just merge it now or i must wait to someone do it?

You can merge it :)

math-araujo reviewed (approved):
LGTM too :)
